### PR TITLE
Updates to support libraries with SCSS and improved developer experience

### PR DESCRIPTION
**Pull request by jensborch**
- Update all JS libraries to last version - e.g. TypeScript 2 
- Added serve:dev task for better developer experience
- Added support for SCSS in libraries
- Added example project for e.g. testing gulp-angular-library-builder
- Fixed support for templates in TypeScript

### CODE REVIEW

**Comment by langecode** [comment]
I have begun looking at this but its a quite substantial pull-request rewriting much of the existing module. From a brief walk through I am not sure I agree with all changes. So it will take some time to test and go through all of  this. It would have been easier had this been chunked into smaller pieces.

**Comment by langecode** [comment]
I would really like the whole `example` directory to go into another pull-request as I think there is a number of issues with that and it has nothing to do with adding type-script support (which I thought this was originally about).

**Comment by jensborch** [comment]
I have created a new pull request without the  `example` directory

File: src/dist.gulp.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-    gulp = require('gulp');
+    gulp = require('gulp'),
+    sequence = require('run-sequence').use(gulp);
 
 module.exports = function (options) {
-    gulp.task('dist', ['build'], function () {
-        return gulp.src([path.join(options.dist, 'build/**/*'), path.join(options.src, '**/*.d.ts'), 'bower.json', 'README.md'])
+
+    gulp.task('dist', function (callback) {
+        sequence('clean', 'build', 'styles-dist', 'dist-copy', callback);
```
Comment (langecode) [inline src/dist.gulp.js]: Please avoid sequences as much as possible as it compromises the natural parallelism of Gulp!

File: src/index.js [hunk 2]
```diff
@@ -7,15 +7,34 @@ module.exports = function (options) {
     options = assign({}, {
         name: 'no-name',
         src: 'src',
-        dist: 'dist'
+        dist: 'dist',
+        target: 'target',
```
Comment (langecode) [inline src/index.js]: Why do you need a `target` directory when `build` is already used as intermediate directory?

**Comment by langecode** [comment]
I am going to decline this pull-request. Each pull-request should be small and focussed there is just way too many concerns in this on request.

Please split this up in at least the following individual requests:
- Example library (not application!!) must adhere to the John Papa Angular 1 style guide ([Angular 1 Style Guide](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md)) as this is what we generally follow and this module was created with that in mind. It is not best practice to keep, e.g., templates in separate directory. Furthermore the Angular component model should be used in favor of directives, etc.
- TypeScript compilation optimization. Seems the code around TypeScript transpiling has been simplified - nice - but should be in separate PR, please.
- Support for SASS, might or might not make sense but is another concern.
- Additional linting
- Handling of test html injecting css, js and more. I do not think this should be part of this module. The intention of this module was to have a build setup for simple Angular libraries not for complete SPA'ish components. I realize you are probably using this for some kind of in-app development testing but it was never the intention of `gulp-angular-library-builder` to build big complex components. If you are in essence building small SPA'ish modules which needs that kind of testing I think you should look at pushing [Gript](https://github.com/Nykredit/gript) in that direction instead. Adding functionality like serving pages, injection of scripts and more will make this simple library builder too much like [Gript](https://github.com/Nykredit/gript) - which was definitely not the intention.

**Pull request closed without merging**